### PR TITLE
PHP7 extension port

### PR DESCRIPTION
**Pull request by jpauli**

### CODE REVIEW

File: ext7/pimple/config.w32 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// $Id$
+// vim:ft=javascript
+
+// If your extension references something external, use ARG_WITH
+// ARG_WITH("pimple", "for pimple support", "no");
+
+// Otherwise, use ARG_ENABLE
+// ARG_ENABLE("pimple", "enable pimple support", "no");
```
Comment (stof) [inline ext7/pimple/config.w32]: shouldn't this be uncommented ?

**Comment by stof** [comment]
duplicating the testsuite looks wrong to me. both versions of the extension should be passing the same testsuite.

and the Travis setup should be updated to compile the extension on PHP 7 too (compiling the PHP 7 one rather than the PHP 5 one).

and what would be the right way to distribute an extension having 2 codebases ? Currently, ext-pimple is not registered on pecl at all, meaning there is no distribution channel yet, but is such a setup supported by pecl easily ? If no, it might be better to have a single entry point changing the included files based on the PHP version

**Comment by jbboehr** [comment]
Closes #184

File: ext7/pimple/README.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+This is Pimple 2 implemented in C
```
Comment (stof) [inline ext7/pimple/README.md]: Pimple 3 actually

File: ext7/pimple/README.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+* PHP >= 7.0
+* Not tested under Windows, might work
```
Comment (stof) [inline ext7/pimple/README.md]: @fabpot it might be worth configuring AppVeyor for that

**Comment by stof** [comment]
The Travis config needs to be updated to compile the PHP 7 extension with PIMPLE_EXT=yes and PHP being at version 7 (we need to add the version 7 too). Currently, adding the PHP 7 in the build matrix would still compile the PHP 5 extension

**Comment by petrkotek** [comment]
@stof / @jbboehr: are there any plans to officially support the extension under PHP7?

**Comment by petrkotek** [comment]
@jpauli do you run this somewhere in production?

**Comment by jpauli** [comment]
No I dont

**Pull request closed without merging**